specification/trafficmanager/resource-manager/Microsoft.Network/stable/2018-04-01/trafficmanager.json

```diff
@@ -0,0 +1,1578 @@
+            "type": "array",
+            "minItems": 2,
+            "maxItems": 2,
+            "collectionFormat": "csv",
+            "items": {
+              "type": "number",
+              "format": "double"
+            }
+          },
+          {
+            "name": "botRight",
+            "in": "query",
+            "required": false,
+            "allowEmptyValue": true,
+            "description": "The bottom right latitude,longitude pair of the rectangular viewport to query for.",
+            "type": "array",
+            "minItems": 2,
+            "maxItems": 2,
+            "collectionFormat": "csv",
+            "items": {
+              "type": "number",
+              "format": "double"
+            }
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The Traffic Manager heatmap.",
+            "schema": {
+              "$ref": "#/definitions/HeatMapModel"
+            }
+          },
+          "default": {
+            "description": "Default response. It will be deserialized as per the Error definition.",
+            "schema": {
+              "$ref": "#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "HeatMap-GET": {
+            "$ref": "./examples/HeatMap-GET.json"
+          },
+          "HeatMap-GET-With-TopLeft-BotRight": {
+            "$ref": "./examples/HeatMap-GET-With-TopLeft-BotRight.json"
+          },
+          "HeatMap-GET-With-Null-Values": {
+            "$ref": "./examples/HeatMap-GET-With-Null-Values.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Network/trafficManagerUserMetricsKeys": {
+      "get": {
+        "tags": [
+          "RealUserMetrics"
+        ],
+        "operationId": "TrafficManagerUserMetricsKeys_Get",
+        "description": "Get the subscription-level key used for Real User Metrics collection.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The subscription-level Traffic Manager Real User Metrics key.",
+            "schema": {
+              "$ref": "#/definitions/UserMetricsModel" 
+            }
+          },
+          "default": {
+            "description": "Default response. It will be deserialized as per the Error definition.",
+            "schema": {
+              "$ref": "#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "TrafficManagerUserMetricsKeys-GET": {
+            "$ref": "./examples/TrafficManagerUserMetricsKeys-GET.json"
+          }
+        }
+      },
+      "put": {
```

Comment:
PUT on a collection is not allowed/supported by ARM. Please include the instancename in the URL. 
It should be /subscriptions/{subscriptionId}/providers/Microsoft.Network/trafficManagerUserMetricsKeys/{metricKeyName}. If it does not really have a name, you can put a default or current as the segment. The pattern wont work for ARM in the current shape. Check other resource types in this spec for reference.

Reply:
Should be /subscriptions/{subscriptionId}/providers/Microsoft.Network/trafficManagerUserMetricsKeys/default